File: typescript/scrabble-score/scrabble-score.ts
type Given = { [key: number]: string }
type Derived = { [letter: string]: number }

const given: Given = {
  1: 'AEIOULNRST',
  2: 'DG',
  3: 'BCMP',
  4: 'FHVWY',
  5: 'K',
  8: 'JX',
  10: 'QZ',
} 

const derived: Derived = {}

// convert given map to one-on-one mapping
Object.keys(given).forEach((key) => {
  [...given[Number(key)]].forEach((c) => {
    derived[c] = Number(key)
  })
})

const score = (word: string  = '') => {
  return [...word]
    .reduce((score, c) => score + derived[c.toUpperCase()], 0)
}

export default score

File: csharp/scrabble-score/ScrabbleScore.cs
using System;
using System.Collections.Generic;

public static class ScrabbleScore
{
    public static int Score(string input)
    {
        int score = 0;
        // given points-to-letters mapping
        Dictionary<int, string> givenMap = new Dictionary<int, string>(){
            { 1, "AEIOULNRST"},
            { 2, "DG" },
            { 3, "BCMP" },
            { 4, "FHVWY" },
            { 5, "K" },
            { 8, "JX" },
            { 10, "QZ" },
        };
        // build one-to-one letter-to-points map programmatically
        Dictionary<char, int> oneToOneMap = new Dictionary<char, int>();
        foreach (KeyValuePair<int, string> entry in givenMap) {
            foreach (char c in entry.Value) {
                oneToOneMap.Add(c, entry.Key);
            }
        }
        // look up score in constant time
        foreach (char c in input.ToUpper()) {
            score += oneToOneMap[c];
        }
        return score;
    }
}